dsymbol/src/dsymbol/conversion/first.d

```diff
@@ -148,6 +149,16 @@ final class FirstPass : ASTVisitor
 			processParameters(currentSymbol, dec.returnType,
 					currentSymbol.acSymbol.name, dec.parameters, dec.templateParameters);
 		}
+
+		if (dec.returnType !is null) {
+
+			// Do lookup if return type exists
+			auto app = appender!string();
+			app.formatNode(dec.returnType);
+
+			currentSymbol.typeLookups.insert(TypeLookupsAllocator.instance.make!TypeLookup(
+					istring(app.data), TypeLookupKind.returnType));
```

Comment:
I'm wondering why aren't you using `addTypeToLookups` like in all the other places that define types? I don't think return types should be handled any differently from variable declaration types, other than being stored in a separate field? (and even that might not really be neccessary imo)

Reply:
I can try doing the suggested change, seem like It works, but I'm kinda unsure what happens behind the hood, but all tests passes :)

Reply:
nice, I just assumed that would work from how the other type resolution stuff would work.

I think this change should help support more different type syntaxes (not just simple concrete types) - want to try them out in some tests to avoid them getting regressed from here or should we just go forward with the PR as-is?

Reply:
Sure we can always do more tests, what do you suggest we test?

Reply:
usage ideas would be `Templated!int`, `auto` (should be null/empty then), `NonExistantType`, `int[2]`, `int[]`, `int*`

but I'm not sure what exactly to test for, so we might not really need anything here.

Reply:
I have added some tests that you suggested, are they sufficient?

Reply:
yes I think they are sufficient